### ISSUE TITLE
is there pretrained model for COCO we can download?

### ISSUE THREAD

**Issue by squashking** [bowenc0221/panoptic-deeplab#4]
Hi Bowen,
Thanks for the code. is there pretrained model for COCO we can download?

**Comment by bowenc0221** [comment]
I'm still trying to reproduce the COCO number, I will release the model once I get a similar result. But COCO is already supported, you can play with it if you like :)

**Comment by lxtGH** [comment]
@bowenc0221 Hi Bowen, Would you share the config files and detialed settings on COCO dataset?

**Comment by bowenc0221** [comment]
Here is part of the config you might need to change. I will upload the config file once I can reproduce a reasonable result.

```
DATASET:
  ROOT: "./datasets/coco"
  DATASET: "coco_panoptic"
  NUM_CLASSES: 133
  TRAIN_SPLIT: 'train2017'
  TEST_SPLIT: 'val2017'
  MIN_RESIZE_VALUE: 641
  MAX_RESIZE_VALUE: 641
  RESIZE_FACTOR: 32
  CROP_SIZE: (641, 641)
  MIRROR: True
  MIN_SCALE: 0.5  # change this
  MAX_SCALE: 1.5  # change this
  SCALE_STEP_SIZE: 0.1
  MEAN: (0.485, 0.456, 0.406)
  STD: (0.229, 0.224, 0.225)
  SEMANTIC_ONLY: False
  IGNORE_STUFF_IN_OFFSET: True
  SMALL_INSTANCE_AREA: 4096
  SMALL_INSTANCE_WEIGHT: 3
LOSS:
  SEMANTIC:
    NAME: "cross_entropy"
    IGNORE: 255
TRAIN:
  IMS_PER_BATCH: 12  # change this
  MAX_ITER: 600000  # change this
TEST:
  CROP_SIZE: (641, 641)
```

Note, the loss function for the semantic branch no longer uses hard pixel mining. Because COCO annotation quality is not good.

**Comment by superxiaoying** [comment]
The paper says the crop size is 1025 * 1025 with batch size 64.
Why 641*641? @bowenc0221

**Comment by bowenc0221** [comment]
<img width="739" alt="Screen Shot 2020-07-28 at 10 19 51 AM" src="https://user-images.githubusercontent.com/25308473/88699187-e0f78100-d0bb-11ea-8128-1cb1abb0ed46.png">

@superxiaoying 
There are two settings in the paper. For ResNet and MobileNet we simply use 641x641. 1025x1025 is used to get the best performance.

**Comment by shipra25jain** [comment]
Hello, 
Thank you for the code!
Using the given settings for COCO, with crop size 512*512, batch size 6 and training for 250k iterations gives me extremely low PQ i.e 2.6. Do you have any idea, what I might be missing? Or have you used this version of code to  train panoptic segmentation for COCO? If yes, then what performance did you get on val split?

Thanks

**Comment by bowenc0221** [comment]
@shipra25jain This is the log of my experiment on COCO using the old code, currently I'm working on COCO with Detectron2 version which can get ~34PQ. I will release the model and config via Detectron2 probably after CVPR deadline.

[coco_experiment_log.txt](https://github.com/bowenc0221/panoptic-deeplab/files/5472922/coco_experiment_log.txt)

**Comment by moodlife** [comment]
@bowenc0221 Hi!.  I try to reproduce the paper on COCO using the old code but can't achieve a similar performance. I wonder if  you can release the D2 version of the COCO version. Thank you!

**Comment by bowenc0221** [comment]
@moodlife my PR to D2 is under review right now, so it should be there soon

**Comment by moodlife** [comment]
@bowenc0221 Hi! I train the model in coco using all your configs but for batchsize 64/iterations 200000 but get a ugly result that PQ is only 31.4. I wonder if you can give me some advice.
Sincerely!

**Comment by moodlife** [comment]
> @bowenc0221 Hi! I train the model in coco using all your configs but for batchsize 64/iterations 200000 but get a ugly result that PQ is only 31.4. I wonder if you can give me some advice. And my learning rate is 0.001. Other parameters are all same.
> Sincerely!

**Comment by bowenc0221** [comment]
COCO models are now available, with 34.2 PQ reproduced for ResNet-50 backbone.

**Comment by bowenc0221** [comment]
Please refer to issue #60 for further discussion related to COCO dataset.
